llvm/lib/Target/DirectX/DXILPostOptimizationValidation.cpp

```diff
@@ -50,15 +51,55 @@ static void reportInvalidDirection(Module &M, DXILResourceMap &DRM) {
   }
 }
 
-} // namespace
+static void reportOverlappingError(Module &M, ResourceInfo R1,
+                                   ResourceInfo R2) {
+  SmallString<64> Message;
```

Comment:
64 isn't large enough - even if the names are a single character each this message is 72 characters long, so this is always heap allocated. If we use 128 then even names of 20ish characters should fit in the small size.